src/main/io/serial.c

```diff
@@ -571,13 +571,17 @@ void serialPassthrough(serialPort_t *left, serialPort_t *right, serialConsumer *
         if (serialRxBytesWaiting(left)) {
             LED0_ON;
             uint8_t c = serialRead(left);
+            // Make sure there is space in the tx buffer
+            while (!serialTxBytesFree(right));
```

Comment:
(This will only postpone overflow until both serial buffers are full)

Reply:
@ledvinap This prevents more data being written to the tx buffer until there is space. The buffer will be getting drained by the UART driver so this loop will only stall execution for one byte transmit time (dependant on baud rate). This change is needed because the size of the UART buffers has been reduced which was breaking avrdude.

I'm just completing a switch from specifying SERIAL_DTR resources to using PINIO resources as requested in your review. I'll be honest; I don't like using PINIO as it's less intuitive (and ease of use is really important), but if that's what you guys want I'll do that.

Reply:
Yes, it will effectively double buffer size - both receive and transmit buffers are used.